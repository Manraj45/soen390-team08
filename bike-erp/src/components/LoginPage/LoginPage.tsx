import { Button, Grid, TextField, Typography } from '@material-ui/core'
import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import bike_logo from '../../assets/images/login_bike_logo.png'
import { credential, login } from '../../redux/actions/AccountActions/accountAction'
import useStyles from './LoginPageStyle'

const LoginPage = (props: any) => {
    const classes = useStyles();

    const handleLogin = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault()
        const email: string = event.currentTarget.email.value
        const password: string = event.currentTarget.password.value
        props.login({ email: email, password:password })
    }
    return (
        <div>
            <Grid container spacing={0} direction="row" className={classes.loginPageWrapper}>
                <Grid item xs={12} md={4} className={classes.grid}>
                    <Typography variant="h3">Badob Inc</Typography>
                    <img src={bike_logo} alt="bike_logo" className={classes.logo}></img>
                </Grid>
                <Grid item xs={12} md={4} className={classes.grid} >
                    <div className={classes.login}>
                        <Typography variant="h5">Already Have An Account</Typography>
                        {
                            props.account.error ? <Typography className={classes.error}>Incorrect Email or Password</Typography> : <></>
                        }
                        <form autoComplete="off" onSubmit={handleLogin}>
                            <TextField name="email" label="Email" className={classes.textfield}></TextField>
                            <br />
                            <TextField type="password" name="password" label="Password" className={classes.textfield} ></TextField>
                            <br />
                            <Button type="submit" variant="contained" color="primary" className={classes.button}>Login</Button>
                            
                            
                            
                            {
                                props.account.access_token ?<Typography>Login Success</Typography>:<></>
                            }
                            
                        </form>
                    </div>
                </Grid>
                <Grid item xs={12} md={4} className={classes.grid}>
                    <div className={classes.register}>
                        <Typography variant="h5">Don't have an account?</Typography>
                        <Button variant="contained" color="primary" className={classes.button}><Link to="/register">Register</Link></Button>
                    </div>
                </Grid>
            </Grid>
        </div >
    )
}
const mapStateToProps = (state: any) => {
    return {
        account: state.account
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        login: (credential: credential) => dispatch(login(credential))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage)
