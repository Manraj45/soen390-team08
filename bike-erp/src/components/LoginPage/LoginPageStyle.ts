import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
    logo: {
        width: "100%"
    },
    loginPageWrapper: {
        [theme.breakpoints.down('sm')]: {
            margin: "0",
            position: "absolute",
            top: "10%"
        },
        [theme.breakpoints.up('md')]: {
            margin: "0",
            position: "absolute",
            top: "30%"
        }
    },
    textfield: {
        width: "80%",
        height: "100%",
        marginBottom: "10px"
    },
    grid: {
        [theme.breakpoints.down('sm')]: {
            marginTop: "50px"
        },
        [theme.breakpoints.up('md')]: {
            marginTop: "10px"
        }
    },
    button: {
        width: "80%",
        marginTop: "10px"
    },
    login: {
        marginTop: "10%"
    },
    register: {
        [theme.breakpoints.down('sm')]: {
            marginTop: "20px",
            marginBottom: "30px"
        },
        [theme.breakpoints.up('md')]: {
            marginTop: "20%"
        },
    },
    error: {
        color: "red"
    }
}))

export default useStyles