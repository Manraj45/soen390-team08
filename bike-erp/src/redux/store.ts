import { applyMiddleware, createStore } from "redux";
import rootReducer from "./rootReducer";
import thunk from "redux-thunk";

// Create store and apply thunk for API request
const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
