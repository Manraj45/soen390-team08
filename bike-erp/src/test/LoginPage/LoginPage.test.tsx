import React from 'react'
import LoginPage from '../../components/LoginPage/LoginPage'
import { shallow, configure } from 'enzyme'
import store from '../../redux/store'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
import { BrowserRouter, BrowserRouter as Router } from 'react-router-dom';
configure({ adapter: new Adapter() });

const setUp = () => {
    const testStore = store
    const wrapper = shallow(<LoginPage store={testStore} />).childAt(0).dive()
    return wrapper
}

describe('LoginPage Component Test', () => {
    let wrapper: any;
    beforeEach(() => {
        wrapper = setUp()
    })

    it("render login page without crashing", () => {
        const component = wrapper.find('#loginPage')
        expect(component.length).toBe(1)
    })

    it('match rendered component with snapshot',()=>{
        const tree = renderer.create(
        <BrowserRouter>
            <LoginPage store={store}/>).toJSON();
        </BrowserRouter>)
        expect(tree).toMatchSnapshot()
    })
    
})

