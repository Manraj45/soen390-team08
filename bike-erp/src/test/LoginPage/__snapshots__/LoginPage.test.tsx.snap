// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component Test match rendered component with snapshot 1`] = `
Array [
  <div
    id="loginPage"
  >
    <div
      className="MuiGrid-root makeStyles-loginPageWrapper-2 MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-grid-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3"
        >
          Badob Inc
        </h3>
        <img
          alt="bike_logo"
          className="makeStyles-logo-1"
          src="login_bike_logo.png"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          className="makeStyles-login-6"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Already Have An Account
          </h5>
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-textfield-3"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
              >
                Email
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
            <br />
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-textfield-3"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
              >
                Password
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="password"
                />
              </div>
            </div>
            <br />
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Login
              </span>
            </button>
          </form>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          className="makeStyles-register-7"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Don't have an account?
          </h5>
          <a
            href="/register"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
                "textDecoration": "none",
              }
            }
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Register
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  ").toJSON();",
]
`;

exports[`LoginPage Component Test match snapshot 1`] = `
Array [
  <div
    id="loginPage"
  >
    <div
      className="MuiGrid-root makeStyles-loginPageWrapper-2 MuiGrid-container"
    >
      <div
        className="MuiGrid-root makeStyles-grid-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3"
        >
          Badob Inc
        </h3>
        <img
          alt="bike_logo"
          className="makeStyles-logo-1"
          src="login_bike_logo.png"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          className="makeStyles-login-6"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Already Have An Account
          </h5>
          <form
            autoComplete="off"
            onSubmit={[Function]}
          >
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-textfield-3"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
              >
                Email
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                />
              </div>
            </div>
            <br />
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-textfield-3"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
              >
                Password
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  name="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="password"
                />
              </div>
            </div>
            <br />
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="submit"
            >
              <span
                className="MuiButton-label"
              >
                Login
              </span>
            </button>
          </form>
        </div>
      </div>
      <div
        className="MuiGrid-root makeStyles-grid-4 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
      >
        <div
          className="makeStyles-register-7"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5"
          >
            Don't have an account?
          </h5>
          <a
            href="/register"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
                "textDecoration": "none",
              }
            }
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Register
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  ").toJSON();",
]
`;
